languages/python/django-oso/django_oso/partial.py

```diff
@@ -41,28 +41,29 @@ def partial_to_query_filter(partial: Expression, model: Model, **kwargs):
 
 
 COMPARISONS = {
-    "Unify": lambda q, f, v: Q(**{f: v}),
-    "Eq": lambda q, f, v: Q(**{f: v}),
-    "Neq": lambda q, f, v: ~Q(**{f: v}),
-    "Geq": lambda q, f, v: Q(**{f"{f}__gte": v}),
-    "Gt": lambda q, f, v: Q(**{f"{f}__gt": v}),
-    "Leq": lambda q, f, v: Q(**{f"{f}__leq": v}),
-    "Lt": lambda q, f, v: Q(**{f"{f}__lt": v}),
+    "Unify": lambda f, v: Q(**{f: v}),
+    "Eq": lambda f, v: Q(**{f: v}),
+    "Neq": lambda f, v: ~Q(**{f: v}),
+    "Geq": lambda f, v: Q(**{f"{f}__gte": v}),
+    "Gt": lambda f, v: Q(**{f"{f}__gt": v}),
+    "Leq": lambda f, v: Q(**{f"{f}__leq": v}),
+    "Lt": lambda f, v: Q(**{f"{f}__lt": v}),
+    "In": lambda f, v: Q(**{f"{f}__in": v}),
```

Comment:
Why is `in` in here if it's not handled by the `compare_expr` function?

Reply:
We use it in `in_expr` as `return COMPARISONS["In"]("pk", subquery)`. It's doing the same thing as the other comparison operators, so seemed odd to have a different mechanism. But we could consider weakening the name to something like `OPERATORS` or `FILTERS` to include non-comparisons. As I write this, I'm actually tempted to add entries with keys `True` and `False` that return our always-true filter `Q(pk__in=[])` and its negation.

Reply:
Gabe convinced me that constants were better: de4b867

Reply:
I think we should put this lambda somewhere else (just inside `in_expr`?) because of the `if expr.operator in COMPARISONS` branch in `translate_expr`.  That set of conditions is subtly ordering dependent.  If the `compare_expr` was moved before the `in_expr` branch, the program would be incorrect.

I also like the ability to have `assert expr.operation in COMPARISONS` in `compare_expr` to make sure the caller passed in a valid expression.  If we add the `In` key to `COMPARISONS` that assert is no longer valid (`In` is not a valid input to `compare_expr`).

Reply:
Ok, f17fdeee adds a top-level function *next to* `COMPARISONS`. I would very much like to keep all of the `Q`-constructing functions together and with the same interface.